src/js/components/Field/Field.js

```diff
@@ -0,0 +1,46 @@
+import React, { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+import { getClassName } from '../../utils/helpers'
+
+class Field extends PureComponent {
+  renderSuccess = () => (
+    <span className='FormGroup-validation'>
+      <Icon name='check' />
+    </span>
+  )
+
+  render () {
+    const { children, status, className, mods, style } = this.props
+
+    const fieldClasses = getClassName(
+      className,
+      status === 'error' && 'is-notValid',
+      status === 'success' && 'is-valid',
+      mods
+    )
+
+    return (
+      <div className={ fieldClasses } style={ style }>
+        { status === 'success' && this.renderSuccess() }
+        { children }
+      </div>
+    )
+  }
+}
+
+Field.propTypes = {
+  children: PropTypes.node.isRequired,
+  status: PropTypes.oneOf([null, 'success', 'error']),
+  className: PropTypes.string,
+  mods: PropTypes.string,
+  style: PropTypes.object
+}
+
+Field.defaultProps = {
+  status: null,
+  className: 'FormGroup',
```

Comment:
I'm not sure what the solution is.. but this throws me off: `FormGroup` is the CSS pattern class used for `Field` but then there is a separate `FieldGroup` component that is pulling in `Field`. Could `Field` and `FieldGroup` be combined maybe to just `FieldGroup`? Or `FormGroup` even? I'm open to changing the CSS component to `FieldGroup` alternately... or are you intentionally keeping the naming convention different since the component structure isn't parallel to the CSS pattern structure?

Reply:
So agree this could look a little confusing and here is where the thought came from.  😬 

The idea of FormGroup, I found very confusing since this is pretty much just a Field 'wrapper' around various inputs that includes the appropriate spacing, labels and messages.  FormGroup to me sounds like a grouping of Form data or multiple fields.

The idea with FieldGroup (currently) is a react helper that wraps the components needed for a field but it has some limitations (currently) that only allow the default settings.  if you want more advances settings, styles, etc you would want to craft the appropriate Field components on your own.

Reply:
I'm cool with the name `FieldGroup` instead.. `FormGroup` is what Bootstrap uses for this pattern so I just stole that name for lack of a better idea. I'd like to update the CSS to use this same name to prevent confusion.

Interesting distinction between `Field` and `FieldGroup`... I get it though. I guess the reality is that there won't always be a strict 1:1 comparison of React components to CSS patterns.

Reply:
yeah, maybe so.  I think we will be pretty close on the core classes though.  We should find some time and talk through these a bit.